src/Field.php

```diff
@@ -322,6 +322,16 @@ public function normalize($value)
                     throw new ValidationException([$this->name => 'Must not be a zero']);
                 }
                 break;
+            case 'email':
+                // remove all line-ends and trim
+                $value = trim(str_replace(["\r", "\n"], '', $value));
+                if ($this->required && empty($value)) {
```

Comment:
This check here probably is not required because it's already validated somewhere else. See line 256 above.
Also see line 264 above, need to add 'email' there too.

Reply:
Also adding it in docs would be nice.

Reply:
1. The check you see is also done for 'string' type (lines 275-280) and there's also a similar validation in 'text' type (282-287) so it may need to be included. The 256 line check just checks for required and empty but without trimming whitespace or removing possible newlines.
2. I've added email to scalar validation.

Reply:
I've added documentation.

In the 'additional documentation' commit I've also arranged field types alphabetically. In that commit, the email addition is in line 115. I think that certain things in the docs should be listed alphabetically because it's more intuitive and easier to spot what you're searching for.